### PR TITLE
Add support for kotlin-allopen plugin

### PR DESCRIPTION
**Pull request by dodgex**
I should have applied the `@KotlinOpen` annotation to all annotations that override a class or method.

### CODE REVIEW

**Comment by dodgex** [comment]
I also updated the kotlin sample project to use the latest kotlin version and the allopen plugin. :)

**Comment by dodgex** [comment]
Aaaaaaaaand... I just realized I missed the Plugin annotations once again... :) I have to leave now, but will add them as soon as I'm back home. :D

**Comment by WonderCsabo** [comment]
Hmm, maybe it is not a good idea, because users not having this plugin will face problems. 

Lets create a method, then, which is not `open`, and uses this mechanism, and add KotlinDoc to it explaining this.

File: examples/kotlin/src/main/kotlin/org/androidannotations/gradle/activity/HelloAndroidActivity.kt [hunk 1]
```diff
@@ -43,3 +44,19 @@ open class HelloAndroidActivity : Activity() {
         helloTextView.text = helloMessage
     }
 }
+
+/**
+ * As of Kotlin 1.0.6+ and AndroidAnnotations 4.4.0, you can use the kotlin-allopen plugin
+ * to remove the need to explicitly declare enhanced classes or methods as <code>open</code>.
+ *
+ * See the build.gradle file or [the Kotlin Documentation](https://kotlinlang.org/docs/reference/compiler-plugins.html#all-open-compiler-plugin)
```
Comment (WonderCsabo) [inline examples/kotlin/src/main/kotlin/org/androidannotations/gradle/activity/HelloAndroidActivity.kt]: I would rather link to our kotlin suppor twiki page. We are not having one, but we should open it. :)
(We should list the `KotlinOpen` annotation there for example.)

Also we could add some gotchas for Kotlin, like `lateinit` and `@JvmField`.

**Comment by dodgex** [comment]
I added a link to https://github.com/androidannotations/androidannotations/wiki/Kotlin 😉 instead of the kotlin docs

File: AndroidAnnotations/androidannotations-core/androidannotations-api/src/main/java/org/androidannotations/api/KotlinOpen.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Utility annotation to easily support kotlin-allopen plugin
+ *
+ * @see "https://kotlinlang.org/docs/reference/compiler-plugins.html#all-open-compiler-plugin"
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations-api/src/main/java/org/androidannotations/api/KotlinOpen.java]: Lets add the AA wiki here as well.

Comment (dodgex) [reply AndroidAnnotations/androidannotations-core/androidannotations-api/src/main/java/org/androidannotations/api/KotlinOpen.java]: Done. I replaced the link and added it to the wiki page

**Comment by WonderCsabo** [comment]
Thanks!

**Pull request merged**